lib/prorate/throttle.rb

```diff
@@ -17,18 +30,30 @@ def throttle!
       identifier = [name, discriminator].join(':')
       
       redis.with do |r|
-        logger.info { "Checking throttle block %s" % name }
-        raise Throttled.new(block_for) if Prorate::BlockFor.blocked?(id: identifier, redis: r)
-
         logger.info { "Applying throttle counter %s" % name }
-        c = Prorate::Counter.new(redis: r, id: identifier, logger: logger, window_size: period)
-        after_increment = c.incr
-        if after_increment > limit
+        resp = run_lua_throttler(redis: r, identifier: identifier, bucket_capacity: limit, leak_rate: @leak_rate, weight: 1, block_for: block_for)
+
+        if resp != "OK"
           logger.warn { "Throttle %s exceeded limit of %d at %d" % [name, limit, after_increment] }
-          Prorate::BlockFor.block!(redis: r, id: identifier, duration: block_for)
-          raise Throttled.new(period)
+          raise Throttled.new("Throttled, please lower your temper and try again in #{resp} seconds")
```

Comment:
The `retry_in_seconds` on the exception that gets raised is actually important, because it's value can then be used for the `Retry-After` header value.